Lesson3/Task3/Model.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.IO;
+using System.IO.Compression;
+
+namespace Task3
+{
+    class Model
+    {
+        public string FilePath { get; private set; }
+        public bool IsPath { get; private set; }
+
+        public string CompressFile()
+        {
+            FileStream source = new FileStream(FilePath, FileMode.Open);
+            FileStream destination = File.Create(Path.GetDirectoryName(FilePath) + @"\archive.zip");
+
+            GZipStream compressor = new GZipStream(destination, CompressionMode.Compress);
+
+            int inputByte = source.ReadByte();
+            while (inputByte != -1)
+            {
+                compressor.WriteByte((byte)inputByte);
+                inputByte = source.ReadByte();
+
+            }
+
+            compressor.Close();
+
+            return destination.Name;
+        }
+
+        public string ShowFileContent()
+        {
+            StreamReader streamReader = File.OpenText(FilePath);
+            string content = streamReader.ReadToEnd();
+            streamReader.Close();
+
+            return content;
+        }
+
+        public bool SearchFile(string targetDirectory, string fileName)
+        {
+            DirectoryInfo dir = new DirectoryInfo(targetDirectory);
+            FileInfo[] fileInfo;
+
+            if (!dir.Exists)
+            {
+                return false;
+            }
+
+            fileInfo = dir.GetFiles(fileName);
```

Comment:
Your solution is great for learning, but `DirectoryInfo` also contains methods for recursive search:
`public FileInfo[] GetFiles(string searchPattern, SearchOption searchOption);`
```
    public enum SearchOption
    {
        //
        // Summary:
        //     Includes only the current directory in a search operation.
        TopDirectoryOnly = 0,
        //
        // Summary:
        //     Includes the current directory and all its subdirectories in a search operation.
        //     This option includes reparse points such as mounted drives and symbolic links
        //     in the search.
        AllDirectories = 1
    }
```